lib/identity_key/identity_key.c

```diff
@@ -0,0 +1,195 @@
+	}
+
+	status = psa_export_key(key_handle, key_buff, IDENTITY_KEY_SIZE_BYTES, &olen);
+	if (status != PSA_SUCCESS) {
+		IDENTITY_KEY_PRINT_VAL("psa_export_key failed! Error: ", status);
+		return -ERR_IDENTITY_KEY_EXPORT_FAILED;
+	}
+
+	if (olen != IDENTITY_KEY_SIZE_BYTES) {
+		IDENTITY_KEY_PRINT_VAL("Exported key size failed! Key size: ", olen);
+		return -ERR_IDENTITY_KEY_EXPORT_FAILED;
+	}
+
+	status = psa_destroy_key(key_handle);
+	if (status != PSA_SUCCESS) {
+		IDENTITY_KEY_PRINT_VAL("psa_destroy_key failed! Error: ", status);
+		return -ERR_IDENTITY_KEY_INTERNAL_ERROR;
+	}
+
+	return 0;
+}
+
+void identity_key_write_random(void)
+{
+	uint8_t key_buff[IDENTITY_KEY_SIZE_BYTES];
+	int err;
+
+	err = generate_random_secp256r1_private_key(key_buff);
+	if (err != 0) {
+		IDENTITY_KEY_PRINT_VAL("The identity key generation failed with error code: ", err);
+		/* Panic raised, as we can't generate the identity key! */
+		IDENTITY_KEY_PANIC();
+	}
+
+	/* Write the key into KMU using the MKEK as encryption key */
+	identity_key_write_key(key_buff);
+
+	/* Clear the unencrypted key from RAM for security reasons */
+	memset(key_buff, 0x0, IDENTITY_KEY_SIZE_BYTES);
+}
+
+void identity_key_write_key(uint8_t key[IDENTITY_KEY_SIZE_BYTES])
+{
+	int err;
+	err = nrf_cc3xx_platform_identity_key_store(NRF_KMU_SLOT_KIDENT, key);
+
+	if (err != NRF_CC3XX_PLATFORM_SUCCESS) {
+		IDENTITY_KEY_PRINT_VAL("The identity key writing to: ", NRF_KMU_SLOT_KIDENT);
+		IDENTITY_KEY_PRINT_VAL("failed with error code: ", err);
+		/* Panic raised, as we can't write the generated identity key! */
+		IDENTITY_KEY_PANIC();
+	}
+}
+
+/* This key is used by the TF-M regression tests and it is taken from the TF-M
+ * repository. It should ONLY be used for testing and debugging purposes.
+ *
+ * TF-M repo path: platform/ext/common/template/tfm_initial_attestation_key.pem
+ * 
+ * Intentionally non-static to ensure it is stored in RAM and not FLASH
+ * (limitation in CryptoCell peripheral.)
+ *
+ * #######  DO NOT USE THIS KEY IN PRODUCTION #######
+ */
+static uint8_t dummy_identity_secp256r1_private_key[IDENTITY_KEY_SIZE_BYTES] = {
+	0xA9, 0xB4, 0x54, 0xB2, 0x6D, 0x6F, 0x90, 0xA4,
+	0xEA, 0x31, 0x19, 0x35, 0x64, 0xCB, 0xA9, 0x1F,
+	0xEC, 0x6F, 0x9A, 0x00, 0x2A, 0x7D, 0xC0, 0x50,
+	0x4B, 0x92, 0xA1, 0x93, 0x71, 0x34, 0x58, 0x5F
+};
+
+void identity_key_write_dummy(void)
+{
+	identity_key_write_key(dummy_identity_secp256r1_private_key);
+}
+
+bool identity_key_mkek_is_written(void)
+{
+	return hw_unique_key_is_written(HUK_KEYSLOT_MKEK);
+}
+
+bool identity_key_is_written(void)
+{
+	int perm;
+
+	for(int i = 0; i < 2; i++)
+	{
+		NRF_KMU->SELECTKEYSLOT = KMU_SELECT_SLOT(NRF_KMU_SLOT_KIDENT + i);
+
+		perm = nrfx_nvmc_uicr_word_read(&NRF_UICR_S->KEYSLOT.CONFIG[NRF_KMU_SLOT_KIDENT + i].PERM);
+
+		NRF_KMU->SELECTKEYSLOT = 0;
+
+		/* Ensure that the KMU slot permissions is set in such a way that the
+		 * key can be read from the CPU and it is not pushable.
```

Comment:
```suggestion
		/* Ensure that the KMU slot permissions is set in such a way that the
		 * key can be read from the CPU and it is not pushable.
```

The identity_key library is rightly relying on NRF_CC3XX_PLATFORM to know of and document what the permissions for the identity key are so these two lines should be deleted.

The business logic for this function would be valid even if this comment was wrong and the permissions were different.